modules/internal/ChapelArray.chpl

```diff
@@ -1078,6 +1078,37 @@ module ChapelArray {
       for i in _value.dimIter(d, ind) do yield i;
     }
 
+   /* Returns a tuple of integers describing the size of each dimension.
+      For a sparse domain, returns the shape of the parent domain.*/
+    proc shape where isRectangularDom(this) {
+      var shape: rank*(int);
+      for (i, r) in zip(1..shape.size, dims()) do
+        shape(i) = r.size;
+      return shape;
+    }
+
+    pragma "no doc"
+    proc shape where isSparseDom(this) {
+      var shape: this._value.parentDom.rank*(int);
+      for (i, r) in zip(1..shape.size, this._value.parentDom.dims()) do
+        shape(i) = r.size;
```

Comment:
I think the sparse version can be the same as the Rectangular one since dims() for a sparse domain returns the dimensions of the parent domain.